### ISSUE TITLE
Team Names 

### ISSUE THREAD

**Issue by mrg1999** [BurntSushi/nfldb#124]
I've got my spreadsheet working very well for 2015 using stats from the nfldb, now I'm looking at next year and beyond to set up a spreadsheet at the beginning of the season.
I'd like to retrieve every NFL team name, abbreviation, respective conference and division from the database. I'm having trouble and would appreciate some help.
Looking to forward seasons, teams change (i.e. Houston Oilers) and I would like to be able to automatically update my spreadsheet accordingly.
In June or July of 2017 I'd like to be able to pull the new info from the database should, say the Raiders move to LA.

**Comment by andr3w321** [comment]
Write some python functions?  There's only 32 teams.  It should not take very long to write some functions ie

```
get_team_city(team_abbr, season_year):
if team_abbr == 'HOU'
    if season_year >= 2016:
        return 'Los Angeles'
    else:
        return 'Houston'
```

**Comment by mrg1999** [comment]
And what if in 2018 the NFL decided to award 8 more franchises (one for each division in the conferences)?
Its not out of the question as the demand certainly is there.
My basic question is can I pull all the team names and their respective conferences and divisions from the nlfdb for a given year?

**Comment by andr3w321** [comment]
There have been zero division or conference changes since nfldb only goes back to 2009.

```
''' Given a team name return its division '''
def get_division(team):
    divisions = [
        {'name': 'afc_east', 'teams': ['NE', 'BUF', 'NYJ', 'MIA']},
        {'name': 'afc_north', 'teams': ['CIN', 'PIT', 'BAL', 'CLE']},
        {'name': 'afc_south', 'teams': ['IND', 'HOU', 'JAC', 'TEN']},
        {'name': 'afc_west', 'teams': ['DEN', 'KC', 'OAK', 'SD']},
        {'name': 'nfc_east', 'teams': ['NYG', 'WAS', 'PHI', 'DAL']},
        {'name': 'nfc_north', 'teams': ['MIN', 'GB', 'CHI', 'DET']},
        {'name': 'nfc_south', 'teams': ['CAR', 'NO', 'ATL', 'TB']},
        {'name': 'nfc_west', 'teams': ['ARI', 'STL', 'SEA', 'SF']}
    ]
    for division in divisions:
        if team in division['teams']:
            return division['name']
            break
    print "Unknown division for {}".format(team)
    return 'UNK'

```

**Comment by BurntSushi** [comment]
> My basic question is can I pull all the team names and their respective conferences and divisions from the nlfdb for a given year?

No. You can see what the DB can give you by looking at nfldb's entity-relationship diagram: http://burntsushi.net/stuff/nfldb/nfldb-condensed.pdf

The key guarantee is that for every play, it has a team that has a corresponding entry in the `team` table. Each team has a unique id, a name and a city. Selecting every team from the `team` table may produce teams that no longer exist! (This isn't true today, since the number and name of teams that existed in 2009 hasn't changed. But it could in the future.)

In other words, you'll probably need to look elsewhere for divisions and such. The data is small if you only care about current info, in which case, @andr3w321's proposal seems reasonable. If you need team/conference info for every year, then I suspect you'll unfortunately have to scrape it from somewhere.

**Comment by mrg1999** [comment]
Andrew,Thanks again for your help.Can I query the 'Team' table and if so, what parameters do I need to pass?I don't know why I'm having so much trouble with this, I've been programming since the 60's but the light just hasn't gone off in my head yet!I'm not understanding the E-R diagram, I tried passing the gamekey to the table query but that gives me an error.
import nfldb
db = nfldb.connect()q = nfldb.Query(db)
q.game(season_year=2015, season_type='Regular', week=12)
for g in q.as_games():
    gk = (g).gamekey
    q.team(gk)
Thanks,Gordon

**Comment by BurntSushi** [comment]
@mrg1999 You'll have to use SQL. You can see an example here: https://github.com/BurntSushi/nfldb/blob/master/nfldb/types.py#L488-L494
